src/watchmaker/config/__init__.py

```diff
@@ -126,8 +127,27 @@ def get_configs(system, worker_args, config_path=None):
         "Command-line arguments merged into worker configs: %s", worker_args
     )
 
+    validate_computer_name_pattern(config)
+
     if not is_valid(config_status):
         log.error("Status config is invalid %s", config_status)
         raise WatchmakerError("Status config is invalid %s" % config_status)
 
     return config, config_status
+
+
+def validate_computer_name_pattern(config):
+    """Ensure pattern is valid and will match entire computer name."""
+    computer_name_pattern = (
+        config.get("salt", {})
+        .get("config", {})
+        .get("computer_name_pattern", {})
+    )
+
+    if computer_name_pattern:
+        try:
+            re.compile(computer_name_pattern)
+        except re.error:
+            raise WatchmakerError(
```

Comment:
```suggestion
            raise InvalidComputerNameError(
```

Reply:
Oh nevermind. We don't know that the pattern is invalid yet here, just that the regex is not valid.